src/storage/storage-cache.ts

```diff
@@ -243,6 +260,62 @@ export class StorageCache {
     };
   }
 
+  // OVERWRITE
+
+  overwriteAllDocsByAuthor(keypair: AuthorKeypair): number | ValidationError {
+    if (this._storage.isClosed()) {
+      throw new StorageIsClosedError();
+    }
+
+    this._storage.overwriteAllDocsByAuthor(keypair);
+
+    let docsToOverwrite = new Set(
+      Array.from(this._docCache.values()).flatMap((cache) => {
+        return cache.docs.filter((doc) => doc.author === keypair.address);
+      })
+    );
+
+    let numOverwritten = 0;
+
+    console.log({ docsToOverwrite });
+
+    for (let doc of docsToOverwrite) {
+      if (doc.content.length === 0) {
+        continue;
+      }
+
+      // remove extra fields
+      let cleanedResult = this._storage.formatValidator.removeExtraFields(doc);
+      if (isErr(cleanedResult)) {
+        return cleanedResult;
+      }
+      let cleanedDoc = cleanedResult.doc;
+
+      // make new doc which is empty and just barely newer than the original
+      let emptyDoc: Doc = {
```

Comment:
I'm a bit nervous about making our own copy of the empty document here, and hoping it turns out the same as the one made by the actual storage.  Everything should be deterministic, but for example I was considering changing the original code to do `timestamp + 2` for some reason, and if we didn't change it here also...

Instead of making instant changes here, we could let the changes trickle in from the actual Storage as they happen.

I'm not 100% sure about this one.

Reply:
Ah, this is what happen in `set` too. I guess I should make `set` call `._storage.ingest` rather than `._storage.set`?

Reply:
Hmmmmm

I guess if the goal is to avoid waiting for the storage, we have to duplicate some code here.  We just need to make sure that we've subscribed to events on the Storage so when it finishes making and saving the document, we update ours to match?

The StorageDriver sets at least one extra property, `_localIndex`, that we can't set ourselves from over here, so that needs to get updated when we hear back from doing the actual set or ingest on the actual Storage.  There might be other extra properties in the future too.

That means there will be a 1 second gap where we've optimistically made our own version of the doc in the cache that doesn't have those properties, which sounds like a hard-to-reproduce bug waiting to happen.

We could ... have an extra property `_pending: true` on our document until we get the official one back?  Gosh, so much complexity

Reply:
> We just need to make sure that we've subscribed to events on the Storage so when it finishes making and saving the document, we update ours to match?

Yeah! For every query that's made to storage cache, a QueryFollower is created. This QueryFollower updates the cache with 'canonical' docs every time its callback fires (https://github.com/earthstar-project/stone-soup/pull/54/files#diff-334c4c7b24030dc0c983c4b00b795748fb0087e05cc4fe445fbe3b397c6a1726R187).

StorageCache can't be used for certain things (like syncing), so I wonder if that gets us a little off the hook regarding potential differences between docs.